File: UniEatsBackEnd/Models/SalesReport.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace UniEatsBackEnd.Models
{
    public class SalesReport
    {
        public int ReportId { get; set; }

        [Required]
        public DateTime StartDate { get; set; }

        [Required]
        public DateTime EndDate { get; set; }

        [Range(0, double.MaxValue, ErrorMessage = "Total sales must be a non-negative value.")]
        public decimal TotalSales { get; set; }

        [Range(0, int.MaxValue, ErrorMessage = "Number of orders must be a non-negative value.")]
        public int NumberOfOrders { get; set; }

        [RegularExpression(@"^(Daily|Weekly|Monthly|Annual)$", ErrorMessage = "Report type must be 'Daily', 'Weekly', 'Monthly', or 'Annual'.")]
        public string ReportType { get; set; }

        public DateTime GeneratedAt { get; set; } = DateTime.Now;

        // Custom validation to ensure end date is greater than or equal to start date
        public bool IsValid()
        {
            return EndDate >= StartDate;
        }
    }
}

File: UniEatsFrontEnd/UniEatsFrontEnd/src/models/sales-report.ts
export interface SalesReport {
  reportId: number; // Unique identifier for the sales report
  startDate: Date; // Start date of the report period
  endDate: Date; // End date of the report period
  totalSales: number; // Total sales in the report period, must be >= 0
  numberOfOrders: number; // Number of orders in the report period, must be >= 0
  reportType: 'Daily' | 'Weekly' | 'Monthly' | 'Annual'; // Type of report
  generatedAt?: Date; // Date when the report was generated, defaults to current date
}
